src/Executor/BodyMultipart.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace OpenRuntimes\Executor;
+
+class BodyMultipart
+{
+    /**
+     * @var array<string, mixed> $parts
+     */
+    private array $parts = [];
+    private string $boundary = "";
+
+    public function __construct(string $boundary = null)
+    {
+        if (is_null($boundary)) {
+            $this->boundary = self::generateBoundary();
+        } else {
+            $this->boundary = $boundary;
+        }
+    }
+
+    public static function generateBoundary(): string
+    {
+        return '-----------------------------' . \uniqid();
+    }
+
+    public function load(string $body): self
+    {
+        $eol =  "\r\n";
+
+        $sections = \explode('--' . $this->boundary, $body);
+
+        foreach ($sections as $section) {
+            if (empty($section)) {
+                continue;
+            }
+
+            if (strpos($section, $eol) === 0) {
+                $section = substr($section, \strlen($eol));
+            }
+
+            if (substr($section, -2) === $eol) {
```

Comment:
Can we add some comments here to make it easier to understand?

Reply:
Ill document it a bit better, and also link to RFC once we move it to utopia/fetch library, probably during QA.
To explain it to you, it removes `--`, which is a force identifier (alongside an additional optional boundary) of a new section in multipart for-data. It's recommended to always add boundaries, too, because the body text had `--`, which would break the parsing.